hazelcast-jet-all/pom.xml

```diff
@@ -127,10 +90,80 @@
                             <artifactId>org.osgi.core</artifactId>
                             <version>4.2.0</version>
                         </additionalDependency>
-
                     </additionalDependencies>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>${maven.bundle.plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>bundle-manifest</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>bundle</goal>
+                        </goals>
+                        <configuration>
+                            <supportedProjectTypes>jar</supportedProjectTypes>
+                            <instructions>
+                                <Embed-Dependency>hazelcast-jet-core;inline=true</Embed-Dependency>
+                                <Embed-Transitive>true</Embed-Transitive>
+                                <Implementation-Title>${project.name}</Implementation-Title>
+                                <Implementation-Version>${project.version}</Implementation-Version>
+                                <Implementation-Vendor-Id>${project.groupId}</Implementation-Vendor-Id>
+                                <Implementation-Vendor>${project.organization.name}</Implementation-Vendor>
+                                <Implementation-URL>${project.url}</Implementation-URL>
+                                <Specification-Title>${project.name}</Specification-Title>
+                                <Specification-Version>${project.artifact.selectedVersion.majorVersion}.${project.artifact.selectedVersion.minorVersion}</Specification-Version>
+                                <Specification-Vendor>${project.organization.name}</Specification-Vendor>
+                                <Bundle-Activator>com.hazelcast.osgi.impl.Activator</Bundle-Activator>
+                                <Main-Class>com.hazelcast.jet.server.StartServer</Main-Class>
+                                <Automatic-Module-Name>com.hazelcast.jet</Automatic-Module-Name>
+                                <Export-Package>
+                                    com.hazelcast.*
+                                </Export-Package>
+                                <Import-Package>
+                                    !org.junit,
```

Comment:
I went through these dependenceis and not all of them are needed here. These are created here if you leave it as `*` 

```com.hazelcast.aws,
javax.annotation,
javax.cache;version="[1.0,2)";resolution:=optional,
javax.cache.configuration;resolution:=optional,
javax.cache.event;version="[1.0,2)";resolution:=optional
javax.cache.expiry;resolution:=optional
javax.cache.integration;resolution:=optional
javax.cache.management;resolution:=optional
javax.cache.processor;resolution:=optional
javax.cache.spi;version="[1.0,2)";resolution:=optional
javax.management
javax.management.openmbean
javax.net.ssl
javax.script
javax.security.auth
javax.security.auth.callback,javax.security.auth.login
javax.transaction.xa,javax.xml.namespace,javax.xml.parsers,javax.xml.transform
javax.xml.transform.dom
javax.xml.transform.sax,javax.xml.transform.stream
javax.xml.validation,javax.xml.xpath,org.apache.log4j,org.apache.log4j.spi,org.apache.logging.log4j
org.apache.logging.log4j.spi,org.osgi.framework
org.slf4j
org.w3c.dom
org.xml.sax
sun.misc```

Reply:
I copied the the hazelcast pom https://github.com/hazelcast/hazelcast/blob/master/hazelcast/pom.xml but I can leave *